src/databases/mysql.rs

```diff
@@ -1,50 +1,114 @@
+use super::databases::AxumDatabasePool;
+use crate::SessionError;
+use async_trait::async_trait;
+use chrono::Utc;
 use sqlx::{pool::Pool, MySql, MySqlPool};
-
+use std::borrow::Cow;
 /// Mysql's Pool type for AxumDatabasePool
 #[derive(Debug, Clone)]
-pub struct AxumDatabasePool(MySqlPool);
-
-impl AxumDatabasePool {
-    /// Grabs the Pool for direct usage
-    pub fn inner(&self) -> &MySqlPool {
-        &self.0
-    }
+pub struct AxumMySqlPool {
+    pool: Pool<MySql>,
 }
-
-impl From<Pool<MySql>> for AxumDatabasePool {
+impl From<Pool<MySql>> for AxumMySqlPool {
     fn from(conn: MySqlPool) -> Self {
-        AxumDatabasePool(conn)
+        AxumMySqlPool { pool: conn }
     }
 }
-
-pub const MIGRATE_QUERY: &str = r#"
-        CREATE TABLE IF NOT EXISTS %%TABLE_NAME%% (
-            `id` VARCHAR(128) NOT NULL,
-            `expires` INTEGER NULL,
-            `session` TEXT NOT NULL,
-            PRIMARY KEY (`id`),
-            KEY `expires` (`expires`)
+#[async_trait]
+impl AxumDatabasePool for AxumMySqlPool {
+    async fn migrate(&self, table_name: &Cow<'static, str>) -> Result<(), SessionError> {
```

Comment:
We should see if we can get &str to work with Cow as that would be a cleaner implementation on our end.